tianshou/highlevel/experiment.py

```diff
@@ -443,18 +544,15 @@ def with_optim_factory(self, optim_factory: OptimizerFactory) -> Self:
 
     def with_optim_factory_default(
         self,
-        betas: tuple[float, float] = (0.9, 0.999),
-        eps: float = 1e-08,
-        weight_decay: float = 0,
+        **kwargs: Unpack[DefaultOptimFactoryParams],
     ) -> Self:
-        """Configures the use of the default optimizer, Adam, with the given parameters.
+        """Configures the use of the default optimizer, with the given parameters.
 
-        :param betas: coefficients used for computing running averages of gradient and its square
-        :param eps: term added to the denominator to improve numerical stability
-        :param weight_decay: weight decay (L2 penalty)
+        :param kwargs: the parameters to use for the optimizer, see `DefaultOptimFactoryParams`.
         :return: the builder
         """
-        self._optim_factory = OptimizerFactoryAdam(betas=betas, eps=eps, weight_decay=weight_decay)
```

Comment:
How is this an improvement?
It seems to me that this just creates confusion as to what the default optimiser and its arguments are.

Reply:
It's an improvement b/c the default values are no longer duplicated. Previously it would have been very easy to change the default values for Adam in the `optim` package without realizing that the default values in the experiment builder would also need to be changed

Reply:
You are solving problems that don't actually exist: The parameters of Adam only change once in a blue moon - and when they do, every responsible programmer will check the usages of `OptimizerFactoryAdam` and update them.

But you have created a new problem - a usability problem. Users get no information on what the default optimizer is or what its parameters are or what they mean. 

Usability is far more important. Why are you optimizing duplication of parameters that literally never change over usability?

Reply:
Dropped the commit and instead added a few block comments to keep the values in sync